src/verifier.h

```diff
@@ -41,30 +44,41 @@ void VerifyWesolowskiProof(integer &D, form x, form y, form proof, uint64_t iter
     }
 }
 
-bool CheckProofOfTimeNWesolowski(integer D, const uint8_t* x_s, const uint8_t* proof_blob, int32_t proof_blob_len, uint64_t iterations, uint64 disc_size_bits, int32_t depth)
-{
+bool CheckProofOfTimeNWesolowskiCommon(integer& D, form& x, const uint8_t* proof_blob, int32_t proof_blob_len, uint64_t& iterations, int depth, int last_segment, bool skip_check = false) {
     int form_size = BQFC_FORM_SIZE;
     int segment_len = 8 + B_bytes + form_size;
     int i = proof_blob_len - segment_len;
-    form x = DeserializeForm(D, x_s, form_size);
-
-    if (proof_blob_len != 2 * form_size + depth * segment_len)
-        return false;
 
     // Loop depth times
     bool is_valid = false;
-    for (; i >= 2 * form_size; i -= segment_len) {
+    for (; i >= last_segment; i -= segment_len) {
         uint64_t segment_iters = BytesToInt64(&proof_blob[i]);
         form proof = DeserializeForm(D, &proof_blob[i + 8 + B_bytes], form_size);
         integer B(&proof_blob[i + 8], B_bytes);
         form xnew;
-        if (VerifyWesoSegment(D, x, proof, B, segment_iters, xnew))
+        if (VerifyWesoSegment(D, x, proof, B, segment_iters, skip_check, xnew))
             return false;
 
         x = xnew;
+        if (segment_iters > iterations) {
+            return false;
+        }
         iterations -= segment_iters;
     }
+    return true;
+}
 
+bool CheckProofOfTimeNWesolowski(integer D, const uint8_t* x_s, const uint8_t* proof_blob, int32_t proof_blob_len, uint64_t iterations, uint64 disc_size_bits, int32_t depth) {
```

Comment:
I left it for `CheckProofOfTimeNWesolowski` and `verify_n_wesolowski`, since those functions are already heavily used both in chiavdf and in chia-blockchain. I removed the field for the 2 new functions.

Reply:
actually I've tracked and removed it from `CheckProofOfTimeNWesolowski`, but I'll still keep it in python bindings function (`verify_n_wesolowski`)